fpsd/features.py

```diff
@@ -0,0 +1,765 @@
+#!/usr/bin/env python3.5
+import pandas as pd
+import os
+from sqlalchemy import create_engine
+from tqdm import tqdm
+import subprocess
+
+
+def compute_bursts(df):
+    """Compute list of burst lengths for a given trace
+    """
+    bursts = []
+    current_burst_length = 0
+
+    for row in df.itertuples():
+        if row.ingoing == False:
+            current_burst_length += 1
+        elif row.ingoing == True and current_burst_length > 0:
+            bursts.append(current_burst_length)
+            # Reset burst length
+            current_burst_length = 0
+
+    # If we were in a burst at the end of the trace, let's save it
+    if current_burst_length > 0:
+        bursts.append(current_burst_length)
+
+    # For some features we need to know the ordering of bursts,
+    # so let's save this information as well
+    ranks = list(range(1, len(bursts) + 1))
+    return bursts, ranks
+
+
+class FeatureStorage():
+    def __init__(self, test_db=False):
+        """Set up database engine"""
+        self.TEST_MODE = test_db
+        if self.TEST_MODE:
+            self.engine = create_engine('postgresql://{}:{}@{}/{}'.format(
+                os.environ['PGUSER'], os.environ['PGPASSWORD'],
+                os.environ['PGHOST'], 'test' + os.environ['PGDATABASE']))
+        else:
+            self.engine = create_engine('postgresql://{}:{}@{}/{}'.format(
+                os.environ['PGUSER'], os.environ['PGPASSWORD'],
+                os.environ['PGHOST'], os.environ['PGDATABASE']))
+        return None
+
+    def execute_query_from_string(self, query):
+        """Method to run a query using self.engine
```

Comment:
I guess I'm not up on my PEP8. Have to say I don't like that particular guideline, but w/e. I guess at some point we should do that for all the docstrings that wrap at 79 elsewhere.